### PR TITLE
test: add Integration tests for hydration and event reply

### PR DESCRIPTION
**Pull request by alan-agius4**
**build: trigger error when size-tracking is enabled but not in limit file**

Trigger an error when attempting to enable size-tracking for a project not listed in the integration-payload.json file.

Before this update, size-checks would succeed even if the project wasn't included in the file size limit.

**test: add Integration tests for hydration and event reply**

This commit introduces integration tests for hydration and event reply functionalities. Additionally, it implements a payload size check for the `event-dispatch-contract.min.js`.

### CODE REVIEW

File: integration/platform-server-hydration/copy-event-dispatch-contract.mjs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import {copyFileSync} from 'node:fs';
```
Comment (devversion) [inline integration/platform-server-hydration/copy-event-dispatch-contract.mjs:1]: Nit: Add comment what this file does, and why it copies into `public/`

**Review by devversion — APPROVED** [review]
LGTM overall

File: goldens/size-tracking/integration-payloads.json [hunk 2]
```diff
@@ -51,5 +51,12 @@
       "polyfills": 33807,
       "defer.component": 371
     }
+  },
+  "platform-server-hydration/browser": {
+    "uncompressed": {
+      "main": 200584,
+      "polyfills": 33807,
+      "event-dispatch-contract.min": 11293
```
Comment (AndrewKushnir) [inline goldens/size-tracking/integration-payloads.json:59]: This number would likely decrease once we land https://github.com/angular/angular/pull/55587 :)

**Review by AndrewKushnir — APPROVED** [review]
@alan-agius4 thanks for creating an integration app and establishing size tracking 👍

Quick comment: we'd probably need to review and tweak thresholds for the JSAction script (in a followup PR). Currently, the tooling triggers an error if the diff is ~5kb or 5% (whichever hits first), [see here](https://github.com/angular/angular/blob/main/scripts/ci/payload-size.js#L18-L19). For JSAction we may want to be more sensitive to the size changes.

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: size-tracking

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 8305e50f8e6cdeb96b226b4c00d086fa4bc647a3.

**Pull request closed without merging**